### DISCUSSION TITLE
Audio data streaming with meta (ICY/Icecast protocol)

### DISCUSSION THREAD

**Discussion by vvvin333** [encode/starlette#2756] [Q&A]
I am trying to implement audio data live streaming with periodical meta inserted in the stream (ICY/Icecast protocol) using StreamingResponse. But while debugging figured out that amount of raw audio bytes sent and received by client (goal is for AudioPlayer, ICY supported) are different. https://stackoverflow.com/q/79142077/15080117
ICY is rather strict protocol with fixed bytes length (**icy-metaint**) and after every this block metadata inserted of **0+** length.
https://thecodeartist.blogspot.com/2013/02/shoutcast-internet-radio-protocol.html
![shoutcast-metadata](https://github.com/user-attachments/assets/404705a1-8f0d-45b2-9b7f-d66669cdb75f)

I looked at the [source code](https://github.com/encode/starlette/blob/7d586f7e39f3e83e8b86636cd41d75e9a879a791/starlette/responses.py#L248) and can see there is `{"type": "http.response.body", "body": chunk, "more_body": True}` message sent format.
Help me please figure out how can I implement custom StreamingResponse to send only raw bytes (just after the first block with headers)?

**Comment by adriangb** [comment]
`StreamingResponse` does stream raw bytes chunk encoded. If you want to send them all in one go (not chunk encoded) then you need to send it all at once.

**Reply by adriangb** [reply]
If you can give a MRE where the "amount of raw audio bytes sent and received by client" is different we may be able to help further.

**Reply by vvvin333** [reply]
Sure. I'll try to simplify my code and debug with more attention. Maybe I am wrong.

**Reply by vvvin333** [reply]
Sorry, I was wrong in my calculation. It is not a problem with starlette
